launcher-patches/0027-proper-fix-for-migrating-profiles-and-same-name-prof.patch

```diff
@@ -0,0 +1,401 @@
++                    var6.setLastVersionId("latest-release");
++                    var1.getProfiles().put(profileUUID, var6);
+                     var1.getProfiles().remove("(Default)");
+-                    var1.setSelectedProfile(var4);
++                    var1.setSelectedProfile(profileUUID);
++                    // olauncher - end
+                 }
+ 
+                 try {
+diff --git a/src/main/java/net/minecraft/launcher/profile/Profile.java b/src/main/java/net/minecraft/launcher/profile/Profile.java
+index a88b0b3..9141d27 100644
+--- a/src/main/java/net/minecraft/launcher/profile/Profile.java
++++ b/src/main/java/net/minecraft/launcher/profile/Profile.java
+@@ -16,6 +16,7 @@ public class Profile implements Comparable<Profile> {
+     public static final Profile.Resolution DEFAULT_RESOLUTION = new Profile.Resolution(854, 480);
+     public static final LauncherVisibilityRule DEFAULT_LAUNCHER_VISIBILITY;
+     public static final Set<MinecraftReleaseType> DEFAULT_RELEASE_TYPES;
++    private transient String uniqueName;
+     private String name;
+     private File gameDir;
+     private String lastVersionId;
+@@ -31,6 +32,7 @@ public class Profile implements Comparable<Profile> {
+     }
+ 
+     public Profile(Profile var1) {
++        this.uniqueName = var1.uniqueName; // olauncher - includes unique name
+         this.name = var1.name;
+         this.gameDir = var1.gameDir;
+         this.playerUUID = var1.playerUUID;
+@@ -142,8 +144,16 @@ public class Profile implements Comparable<Profile> {
+         this.launcherVisibilityOnGameClose = var1;
+     }
+ 
++    public String getUniqueName() {
++        return uniqueName;
++    }
++
++    public void setUniqueName(String uniqueName) {
++        this.uniqueName = uniqueName;
++    }
++
+     public int compareTo(Profile var1) {
+-        return var1 == null ? -1 : this.getName().compareTo(var1.getName());
++        return var1 == null ? -1 : this.getUniqueName().compareTo(var1.getUniqueName());
+     }
+ 
+     static {
+diff --git a/src/main/java/net/minecraft/launcher/profile/ProfileManager.java b/src/main/java/net/minecraft/launcher/profile/ProfileManager.java
+index bb7d8a0..c6c6f4a 100644
+--- a/src/main/java/net/minecraft/launcher/profile/ProfileManager.java
++++ b/src/main/java/net/minecraft/launcher/profile/ProfileManager.java
+@@ -3,33 +3,11 @@ package net.minecraft.launcher.profile;
+ import com.google.common.base.Objects;
+ import com.google.common.collect.Lists;
+ import com.google.common.collect.Maps;
+-import com.google.gson.Gson;
+-import com.google.gson.GsonBuilder;
+-import com.google.gson.JsonDeserializationContext;
+-import com.google.gson.JsonDeserializer;
+-import com.google.gson.JsonElement;
+-import com.google.gson.JsonObject;
+-import com.google.gson.JsonParseException;
+-import com.google.gson.JsonParser;
+-import com.google.gson.JsonSerializationContext;
+-import com.google.gson.JsonSerializer;
++import com.google.gson.*;
```

Comment:
Could you please reconfigure your IDE to not prefer on-demand imports? I don't want MORE conflicts due to import shuffling.